tests/cli/report-interval.test

```diff
@@ -0,0 +1,60 @@
+
+# Input for the following tests:
+
+2019-01-01
+    (a)   2
+
+2019-02-01
+    (a)   1
+
+
+
+
+# The last report interval option takes precedence.
+
+$ hledger -f- register --weekly --monthly
+>
+2019/01                 a                                        2             2
+2019/02                 a                                        1             3
+
+$ hledger -f- register --monthly --weekly
+>
+2018/12/31w01           a                                        2             2
+2019/01/28w05           a                                        1             3
+
+# The last report interval option takes precedence.
+# The --period expression is no exception.
+$ hledger -f- register -p 'monthly in 2019' --weekly
+>
+2018/12/31w01           a                                        2             2
+2019/01/28w05           a                                        1             3
+
+$ hledger -f- register --weekly -p 'monthly in 2019'
+>
+2019/01                 a                                        2             2
+2019/02                 a                                        1             3
+
+
+
+# The next three tests test the bugfix for issue #1008:
+# The report interval from an --period expression only counts if it is
+# explicitly specified (e.g. -p 'monthly in ...').
```

Comment:
Actually it should work like:
"start date, end date, and report interval are calculated from the union of -b/-e/-p/-D/-W/-M/-Q/-Y flags and date: arguments, with those appearing later on the command line taking precedence".

Reply:
PS I don't mean that we should merge multiple date ranges. Here are some examples that hopefully show what I mean:
```
-b 1/1 -p apr    # same as -b 4/1 -e 5/1 (the -p's start date overrides the -b)
-p apr -b 1/1    # same as -b 1/1 -e 5/1 (-b overrides the -p's start date)
```

Reply:
OK, I can add some tests for this. But they would be not directly related to #1008 und this test file `report-interval.test`. So I would add a new file `date-options.test` as you suggested previously.

Regarding your first comment: --period also sets the `ReportInterval` to `NoInterval` if no interval is specified. And this `NoInterval` would override a previous -W/-M/... which does not make sense for me.